_re2.cc

```diff
@@ -889,7 +951,7 @@ _compile(PyObject* self, PyObject* args, PyObject* kwds)
   PyObject *pattern;
 
   if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!:_compile", (char**)kwlist,
-                                   &PyString_Type, &pattern)) {
+                                   REGEX_OBJECT_TYPE, &pattern)) {
```

Comment:
so this can only take a unicode object on py3 and not a bytes object? I saw earlier that you allowed both unicode and bytes in a different `PyArg_ParseTupleAndKeywords` call -- could you explain what's different here?